experimental/waf.go

```diff
@@ -0,0 +1,36 @@
+// Copyright 2024 Juan Pablo Tosso and the OWASP Coraza contributors
+// SPDX-License-Identifier: Apache-2.0
+
+package experimental
+
+import (
+	"context"
+	"strings"
+
+	"github.com/corazawaf/coraza/v3/internal/corazawaf"
+	"github.com/corazawaf/coraza/v3/types"
+)
+
+type Option func(*corazawaf.Options)
```

Comment:
The only reason I use functional args is to avoid `NewSomething(Options{})` when I don't want options, that can be fixed by adding two methods `NewSomething` and `NewSomethingWithOptions` but I guess it is a matter of taste.

Reply:
I personally find two methods to be ok, lean towards consistency with the waf config. If you're interested in changing that to options pattern as well for a potential major then that'd be ok too

Reply:
👍  changed.